### PR TITLE
[issue 168] Bump Infinispan operator CRDs

### LINKED ISSUE #168: Intersmash does not support Infinispan operator latest version
**Is your feature request related to a problem? Please describe.**
Intersmash provisioning tooling currently supports Infinispan operator channel `2.3.x`.
This channel has been the `latest` until recently, and is bound to Infinispan `14.z`. 
The related community integration tests [are currently failing](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/Intersmash_intersmash/165/pull-ci-Intersmash-intersmash-main-intersmash-e2e/1780306553820155904) since the default channel (i.e. `stable`) is now deploying Infinispan `15.x`.
The failing test is signaling that:
- Intersmash implementation for Infinispan provisioning is not aligned with latest community version, but it is aligned with latest productized version (i.e. Red Hat DataGrid, which is based on Infinispan 14.z")
- The Intersmash e2e tests should configure the supported Infinispan Operator channel to match `2.3.x` explicitly

**Describe the solution you'd like**
1. **short-term**: freeze supported InfinispanOperator channel version to `2.3.x`, make e2e tests work again, freeze documentation and be aligned with the support for productized `latest`
2. **medium-term, in parallel**: evaluate the best choice for allowing the provisioning tooling to support `latest` in both configuration, i.e. when provisioning either the community or the productized versions, e.g. by splitting the Infinispan provisioning implementation to support both the product and the community latest versions, in a similar (but most likely more easy) way it's done for Keycloak/RH-SSO.

**Describe alternatives you've considered**
See above

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
Bumping the Infinispan operator CRDs to fix the latest test failures in prod e2e tests.

Fix #168 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - [x] I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java [hunk 1]
```diff
@@ -73,31 +73,28 @@ public void deploy() {
 		ffCheck = FailFastUtils.getFailFastCheck(EventHelper.timeOfLastEventBMOrTestNamespaceOrEpoch(),
 				getApplication().getName());
 		subscribe();
-
-		// create custom resources
-		int replicas = getApplication().getInfinispan().getSpec().getReplicas();
+		// create Infinispan CR
```
Comment (tommaso-borgato) [inline provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:76]: @fabiobrz out of curiosity, are these changes strictly related to the version upgrade or is a general improvement?

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java [hunk 2]
```diff
@@ -144,6 +124,19 @@ public List<Pod> getPods() {
 				: Lists.emptyList();
 	}
 
+	public List<Pod> getInfinispanPods() {
+		return getInfinispanPods(getApplication().getName());
+	}
+
+	public static List<Pod> getInfinispanPods(final String clusterName) {
+		return OpenShiftProvisioner.openShift.inNamespace(OpenShiftConfig.namespace()).pods().list().getItems().stream().filter(
+				p -> p.getMetadata().getLabels().entrySet().stream()
```
Comment (tommaso-borgato) [inline provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:133]: @fabiobrz are these labels assigned by the operator? is there any doc about this?

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java (hunk 1 above)
Comment (fabiobrz) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:76]: The main goal is to bump the version but I pushed some minor fixes/cleanup as well.

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java [hunk 3]
```diff
@@ -107,32 +104,15 @@ public void undeploy() {
 
 	@Override
 	public void scale(int replicas, boolean wait) {
-		StatefulSet statefulSet = OpenShiftProvisioner.openShift.getStatefulSet(getApplication().getName());
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:110]: It is using a stateful set, but the logic for identifying the Infinispan CR pods has changed over the time and the new one - see the new getInfinispanPods()` method - seems the one that aligns to the current operator implementation better.

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java (hunk 2 above)
Comment (fabiobrz) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:133]: You're right, I might need to back up this logic with resources. Thanks, fixing.

Comment (tommaso-borgato) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:133]: @fabiobrz are you adding any comment here? I mean, should I expect one before closing the thread?

File: testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/InfinispanOperatorProvisionerTest.java [hunk 4]
```diff
@@ -115,6 +114,9 @@ public static void removeOperatorGroup() {
 
 	@AfterEach
 	public void customResourcesCleanup() {
+		INFINISPAN_OPERATOR_PROVISIONER.infinispansClient().list().getItems().stream()
```
[the remark this answers is not in the document]
Comment (fabiobrz) [reply testsuite/integration-tests/src/test/java/org/jboss/intersmash/testsuite/provision/openshift/InfinispanOperatorProvisionerTest.java:117]: Not a new CR, I added just one missing delete statement, i.e. the test was not removing `Infinispan` CRs, which resulted in an incomplete cleanup.

File: provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java (hunk 2 above)
Comment (fabiobrz) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:133]: Yep, I'll add one comment, thanks. The only reference about the `app=infinispan-pod` label I've found is in https://infinispan.org/docs/infinispan-operator/main/operator.html - although there it is mentioned just for the `affinity` aspects.

Comment (tommaso-borgato) [reply provisioners/src/main/java/org/jboss/intersmash/provision/openshift/InfinispanOperatorProvisioner.java:133]: Thank you @fabiobrz

**Pull request merged**